http/src/v0/block.rs

```diff
@@ -102,8 +81,106 @@ pub fn put<T: IpfsTypes>(
     path!("block" / "put")
         .and(with_ipfs(ipfs))
         .and(query::<PutQuery>())
-        .and(multipart::form())
-        .and_then(put_query)
+        .and(warp::header::<Mime>("content-type"))
+        .and(warp::body::stream())
+        .and_then(inner_put)
+}
+
+async fn inner_put<T: IpfsTypes>(
+    ipfs: Ipfs<T>,
+    opts: PutQuery,
+    mime: Mime,
+    body: impl Stream<Item = Result<impl Buf, warp::Error>> + Unpin,
+) -> Result<impl Reply, Rejection> {
+    let boundary = mime.get_param("boundary").map(|v| v.to_string()).unwrap();
```

Comment:
FIXME: shouldn't be an unwrap but ... some error. in HTTP it should probably be a client error.